service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/NoticeOfChangeControllerTest.java

```diff
@@ -67,129 +56,256 @@
     @MockBean
     private CaseAssignmentApi caseAssignmentApi;
 
+    @MockBean
+    private NotificationClient notificationClient;
+
     NoticeOfChangeControllerTest() {
         super("noc-decision");
     }
 
-    @BeforeEach
-    void init() {
-        givenFplService();
-        givenSystemUser();
-        when(caseDataApi.getAuditEvents(USER_AUTH_TOKEN, SERVICE_AUTH_TOKEN, false, CASE_ID.toString()))
-            .thenReturn(AUDIT_EVENTS);
-        when(idamClient.getUserByUserId(USER_AUTH_TOKEN, SOLICITOR_ID))
-            .thenReturn(SOLICITOR_USER);
-        when(caseAssignmentApi.applyDecision(eq(USER_AUTH_TOKEN), eq(SERVICE_AUTH_TOKEN), requestCaptor.capture()))
-            .thenReturn(ASSIGNMENT_RESPONSE);
-    }
-
-    @Test
-    void shouldAddRespondentRepresentation() {
-
-        final Element<Respondent> respondent1 = element(Respondent.builder()
-            .party(RespondentParty.builder()
-                .firstName("First")
-                .lastName("Respondent")
-                .build())
-            .build());
-
-        final Element<Respondent> respondent2 = element(Respondent.builder()
-            .party(RespondentParty.builder()
-                .firstName("Second")
-                .lastName("Respondent")
-                .build())
-            .build());
-
-        final Element<Respondent> respondent3 = element(Respondent.builder()
-            .party(RespondentParty.builder()
-                .firstName("Third")
-                .lastName("Respondent")
-                .build())
-            .build());
-
-        final ChangeOrganisationRequest changeRequest = ChangeOrganisationRequest.builder()
-            .organisationToAdd(NEW_ORGANISATION)
-            .caseRoleId(caseRoleDynamicList("[SOLICITORB]"))
+    @Nested
+    class AboutToStart {
```

Comment:
usually we dont split test like this. We have separate class for each type of callback
e.g NoticeOfChangeAboutToStartControllerTest or NoticeOfChangeSubmittedControllerTest